internal/adapters/redisadapters/redisadapters.go

```diff
@@ -0,0 +1,261 @@
+func (r *RedisAdapter) WriteAccessToken(ctx context.Context, accessToken models.AccessToken) error {
+
+	return r.Rdb.HSet(
+		ctx,
+		"accessTokens-"+accessToken.ID,
+		"accessToken",
+		accessToken.Value,
+		"expiresAt",
+		accessToken.ExpiresAt.Unix(),
+		"URL",
+		accessToken.URL,
+		"type",
+		accessToken.Type,
+	).Err()
+}
+
+// WriteRefreshToken writes the associated ID, access token value, expiration and tokenID of a refresh token to Redis
+func (r *RedisAdapter) WriteRefreshToken(ctx context.Context, refreshToken models.RefreshToken) error {
+
+	return r.Rdb.HSet(
+		ctx,
+		"refreshTokens-"+refreshToken.ID,
+		"refreshToken",
+		refreshToken.Value,
+		"expiresAt",
+		refreshToken.ExpiresAt.Unix(),
+	).Err()
+}
+
+// WriteToIndexExpiringTokens writes the associated expiration and tokenID of an access token to Redis
+func (r *RedisAdapter) WriteToIndexExpiringTokens(ctx context.Context, accessToken models.AccessToken) error {
+
+	var z1 redis.Z
+	z1.Score = float64(accessToken.ExpiresAt.Unix())
+	z1.Member = accessToken.ID
+
+	return r.Rdb.ZAdd(
+		ctx,
+		"indexExpiringTokens",
+		z1,
+	).Err()
+}
+
+// WriteProjectToken writes the project ID and associated expiration and tokenID of a project to Redis
+func (r *RedisAdapter) WriteProjectToken(ctx context.Context, projectID int, accessToken models.AccessToken) error {
+
+	z1 := redis.Z{
+		Score:  float64(accessToken.ExpiresAt.Unix()),
+		Member: accessToken.ID,
+	}
+
+	return r.Rdb.ZAdd(
+		ctx,
+		"projectTokens-"+strconv.Itoa(projectID),
+		z1,
+	).Err()
+}
+
+// Remove/delete functions
+
+// RemoveSession removes a session entry from Redis
+func (r *RedisAdapter) RemoveSession(ctx context.Context, sessionID string) error {
+
+	return r.Rdb.Del(
+		ctx,
+		"session-"+sessionID,
+	).Err()
+}
+
+// RemoveAccessToken removes an access token entry from Redis
+func (r *RedisAdapter) RemoveAccessToken(ctx context.Context, accessTokenID string) error {
+
+	return r.Rdb.Del(
+		ctx,
+		"accessTokens-"+accessTokenID,
+	).Err()
+}
+
+// RemoveRefreshToken removes an access token entry from Redis
+func (r *RedisAdapter) RemoveRefreshToken(ctx context.Context, refreshTokenID string) error {
+
+	return r.Rdb.Del(
+		ctx,
+		"refreshTokens-"+refreshTokenID,
+	).Err()
+}
+
+// RemoveFromIndexExpiringTokens removes an access token entry in the indexExpiringTokens sorted set from Redis
+func (r *RedisAdapter) RemoveFromIndexExpiringTokens(ctx context.Context, accessToken models.AccessToken) error {
+
+	var z1 redis.Z
+	z1.Score = float64(accessToken.ExpiresAt.Unix())
+	z1.Member = accessToken.ID
+
+	return r.Rdb.ZRem(
+		ctx,
+		"indexExpiringTokens",
+		z1,
+	).Err()
+}
+
+// RemoveProjectToken removes an access token entry in a projectTokens sorted set from Redis
+func (r *RedisAdapter) RemoveProjectToken(ctx context.Context, projectID int, accessToken models.AccessToken) error {
```

Comment:
Lets leave these as you have set them up here. But note that we will probably need to change these once we start working the service that will provide project tokens to the KG. 

But lets worry about this when we start implementing the project token service. This has the lowest priority of all components. So let's just forget about implementing the project token stuff for the next little bit until we finish all other components.